### ISSUE TITLE
Enhance custom grid snap

### ISSUE THREAD

**Issue by NewCreature** [NewCreature/EOF#99]
_From [raynebc](https://code.google.com/u/raynebc/) on May 22, 2010 16:17:10_

Being able to set a per-measure grid snap would be useful for some folks,
as EOF's current custom grid snap only allows per-beat values.

_Original issue: http://code.google.com/p/editor-on-fire/issues/detail?id=99_

**Comment by NewCreature** [comment]
_From [raynebc](https://code.google.com/u/raynebc/) on May 22, 2010 14:20:51_

r170 starts this issue

Remaining work involves getting it to work properly, as EOF seems to require a
minimum of 1/4 measure grid snap value, and the sustains aren't working quite right
yet.  I'd also like to make the custom grid snap dialog window keep the previous
radio button set ("measure" or "beat") since the last specified number is shown.

**Status:** Started

**Comment by NewCreature** [comment]
_From [raynebc](https://code.google.com/u/raynebc/) on May 22, 2010 14:23:09_

Also, the "per measure" radio button should probably be disabled if the Time
Signature hasn't been set, since a default TS of 4/4 is not currently assumed in EOF.

**Comment by NewCreature** [comment]
_From [raynebc](https://code.google.com/u/raynebc/) on May 23, 2010 15:54:28_

r171 continues this issue

The "measure" and "beat" radio buttons retain the last set value between edits of the
custom grid snap value.

**Comment by NewCreature** [comment]
_From [xander4j...@yahoo.com](https://code.google.com/u/111302640723734240985/) on May 24, 2010 13:30:12_

For this to work you will probably need to find the start and end times of the measure 
in which time 'p' (passed to eof_snap_logic() function) is located. Then you can just 
split up the total time by the specified interval and find the closest snap position 
to 'p'. You would have to apply the same technique in eof_snap_length_logic(). Are you 
still working on this issue?

**Comment by NewCreature** [comment]
_From [raynebc](https://code.google.com/u/raynebc/) on May 24, 2010 13:44:12_

Eventually I might.  It would probably take me longer to figure out than it would
take you.

**Comment by NewCreature** [comment]
_From [xander4j...@yahoo.com](https://code.google.com/u/111302640723734240985/) on May 24, 2010 14:47:41_

The reason I ask is because it is marked as "Started." I didn't want to mess with it 
if you were working on it.

**Comment by NewCreature** [comment]
_From [raynebc](https://code.google.com/u/raynebc/) on May 24, 2010 15:21:59_

Feel free.  I've gotten into a bad habit of leaving an issue in started if I have
worked on it and plan to come back to it later.

**Status:** Accepted

**Comment by NewCreature** [comment]
_From [xander4j...@yahoo.com](https://code.google.com/u/111302640723734240985/) on May 25, 2010 13:23:30_

**Status:** Started

**Comment by NewCreature** [comment]
_From [xander4j...@yahoo.com](https://code.google.com/u/111302640723734240985/) on May 25, 2010 14:14:07_

**Status:** Fixed

**Comment by NewCreature** [comment]
_From [raynebc](https://code.google.com/u/raynebc/) on May 25, 2010 19:49:46_

I found that a very slight issue exists in this a custom grid snap per measure
doesn't track quite right between tempo changes.  For example, if I set a grid snap
of 1/5 measure, and place 5 consecutive 1/5 measure notes, the end of the fifth note
doesn't end at the end of the measure if there are tempo changes within the measure.
 This can be manually corrected by dragging the affected notes to the correct grid
snap position.

**Status:** Accepted
